### PR TITLE
[PROGRAMMES-6403] Clip related programmes

### PR DESCRIPTION
**Pull request by benolot**
Ticket: https://jira.dev.bbc.co.uk/browse/PROGRAMMES-6403

### CODE REVIEW

**Comment by programmes-developers** [comment]
Can one of the admins verify this patch?

**Comment by BjornTwachtmann** [comment]
@votemike that's actually a requirement, because of course it is, https://jira.dev.bbc.co.uk/browse/PROGRAMMES-4690. V2 has it's own presenter for linked data programmes, which is why it's different, but I'm keen to avoid that here.

**Comment by benolot** [comment]
Rebased

File: src/Controller/FindByPid/ClipController.php [hunk 1]
```diff
@@ -21,7 +22,8 @@ public function __invoke(
         AdaClassService $adaClassService,
         GroupsService $groupsService,
         RelatedLinksService $relatedLinksService,
-        StructuredDataHelper $structuredDataHelper
+        StructuredDataHelper $structuredDataHelper,
+        AdaProgrammeService $adaProgrammeService
```
Comment (votemike) [inline src/Controller/FindByPid/ClipController.php]: Maybe move this up next to the other Ada thing? For readabilities sake?

File: src/Controller/FindByPid/ClipController.php [hunk 2]
```diff
@@ -36,21 +38,27 @@ public function __invoke(
 
         $featuredIn = $groupsService->findByCoreEntityMembership($clip, 'Collection');
 
+        $relatedProgrammesPromise = new FulfilledPromise([]);
+        if ($clip->getOption('show_enhanced_navigation')) {
```
Comment (votemike) [inline src/Controller/FindByPid/ClipController.php]: This `if` condition is duplicated below.

File: src/Controller/FindByPid/ClipController.php [hunk 3]
```diff
@@ -36,21 +38,27 @@ public function __invoke(
...
             $relatedTopicsPromise = $adaClassService->findRelatedClassesByContainer($clip, true, 10);
         }
 
         $resolvedPromises = $this->resolvePromises([
             'relatedTopics' => $relatedTopicsPromise,
+            'relatedProgrammes' => $relatedProgrammesPromise,
         ]);
 
-        $parameters = [
+        return $this->renderWithChrome('find_by_pid/clip.html.twig', array_merge($resolvedPromises, [
```
Comment (votemike) [inline src/Controller/FindByPid/ClipController.php]: Do you think this is more readable than the separate $parameters array?

File: src/Controller/FindByPid/ClipController.php [hunk 4]
```diff
@@ -11,6 +11,7 @@
 use BBC\ProgrammesPagesService\Domain\Entity\Episode;
 use BBC\ProgrammesPagesService\Domain\Entity\ProgrammeContainer;
 use BBC\ProgrammesPagesService\Service\RelatedLinksService;
+use App\ExternalApi\Ada\Service\AdaProgrammeService;
```
Comment (votemike) [inline src/Controller/FindByPid/ClipController.php]: Could you run "Optimize Imports" over this file?

**Comment by benolot** [comment]
Test this please

**Pull request merged**